### PR TITLE
Use pull request based automerge

### LINKED ISSUE #172: Renovate not automerging
## Describe the bug

I've changed our Renovate bot configuration.

The intended end-goal is to automatically merge passing `renovate/**` branches into `main` if all tests pass.
This is not working properly for us, right now.

## Give the steps to reproduce

Let's use the lockfile update branch as an example:

1. Renovate creates a new branch, the 3 tests (2 GitHub, 1 Vercel) are passing.
2. Renovate takes 2 runs to automerge, one to create the branch, one to merge.
3. It's now 5 hours later but the lockfile is still not merged.

## Have you thought of a possible solution?

I'll go ask the Renovate maintainers what they think is causing the problem...
I've done what I could here. 😄 

I've opened a Renovate bot help discussion: https://github.com/renovatebot/renovate/discussions/10554

## Do you want to help fix the bug?

Ofcourse! 😄

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Remove `"automergeType": "branch"`
- Drop testing on `renovate/**` branches

## Context

Fixes #172

Turns out `"automergeType": "branch"` does not work when you also require PR approval for merges to `main`.
We also do not need to have the `renovate/**` branch in the workflow matrix, as we revert to using PR based automerge.

I think this should get automerge working again (albeit with the noise of creating the PR).

### CODE REVIEW

**Review by Belco90 — APPROVED** [review]
If I understood correctly, this is reverting the previous config updated recently right?

**Comment by HonkingGoose** [comment]
We're reverting the branch based automerge, as that won't work when the repository requires "PR approval" before merging anything into `main`.

**Pull request merged**
